Assets/HoloToolkit/Input/Scripts/Gaze/GazeManager.cs

```diff
@@ -0,0 +1,349 @@
+            }
+
+            if (GazeTransform == null)
+            {
+                if (Camera.main != null)
+                {
+                    GazeTransform = Camera.main.transform;
+                }
+                else
+                {
+                    Debug.LogError("Gaze Manager was not given a GazeTransform and no main camera exists to default to.");
+                }
+            }
+        }
+
+        private void Update()
+        {
+            if (GazeTransform == null)
+            {
+                return;
+            }
+
+            Vector3 newGazeOrigin = GazeTransform.position;
+            Vector3 newGazeNormal = GazeTransform.forward;
+
+            // Update gaze info from stabilizer
+            if (Stabilizer != null)
+            {
+                Stabilizer.UpdateStability(newGazeOrigin, GazeTransform.rotation);
+                newGazeOrigin = Stabilizer.StablePosition;
+                newGazeNormal = Stabilizer.StableRay.direction;
+            }
+
+            GazeOrigin = newGazeOrigin;
+            GazeNormal = newGazeNormal;
+
+            // Perform raycast to determine gazed object
+            GameObject previousFocusObject = HitObject;
+
+            // If there is only one priority, don't prioritize
+            if (RaycastLayerMasks.Length == 1)
+            {
+                IsGazingAtObject = Physics.Raycast(GazeOrigin, GazeNormal, out hitInfo, MaxGazeCollisionDistance, RaycastLayerMasks[0]);
+            }
+            else
+            {
+                // Raycast across all layers and prioritize
+                RaycastHit? hit = PrioritizeHits(Physics.RaycastAll(new Ray(GazeOrigin, GazeNormal), MaxGazeCollisionDistance, -1));
+
+                IsGazingAtObject = hit.HasValue;
+                if (IsGazingAtObject)
+                {
+                    hitInfo = hit.Value;
+                }
+            }
+            
+            if (IsGazingAtObject)
+            {
+                HitObject = HitInfo.collider.gameObject;
+                HitPosition = HitInfo.point;
+                lastHitDistance = HitInfo.distance;
+            }
+            else
+            {
+                HitObject = null;
+                HitPosition = GazeOrigin + (GazeNormal * lastHitDistance);
+            }
+
+            // If we have a unity event system, perform graphics raycasts as well to support Unity UI interactions
+            if (EventSystem.current != null)
+            {
+                // NOTE: We need to do this AFTER we set the HitPosition and HitObject since we need to use HitPosition to perform the correct 2D UI Raycast.
+                RaycastUnityUI();
+            }
+
+            // Dispatch changed event if focus is different
+            if (previousFocusObject != HitObject && FocusedObjectChanged != null)
+            {
+                FocusedObjectChanged(previousFocusObject, HitObject);
```

Comment:
No, it raises FocusedObjectChanged. You can get the OnFocusEnter and OnFocusExit message through the input module, by implementing the IFocusable interface in your MonoBehaviour